packages/ecc-client-ga4gh-tes/src/components/tes-get-run/definition/tesGetRun.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  FASTElement,
+  attr,
+  customElement,
+  observable,
+} from "@microsoft/fast-element";
+import template from "./tesGetRun.template.js";
+import styles from "./tesGetRun.styles.js";
+import { fetchTask } from "../../../data/index.js";
+import { deleteTask } from "../../../data/Post/tesPost.js";
+
+interface TaskData {
+  id: string;
+  state: string;
+  name: string;
+  description: string;
+  executors: Executor[];
+  logs: Log[];
+  creation_time: string;
+}
+
+interface Executor {
+  image: string;
+  command: string[];
+}
+
+interface Log {
+  logs: LogEntry[];
+  metadata: {
+    USER_ID: string;
+  };
+  start_time: string;
+  end_time: string;
+}
+
+interface LogEntry {
+  start_time: string;
+  end_time: string;
+  stdout: string;
+  exit_code: number;
+}
+
+@customElement({
+  name: "ecc-tes-get-run",
+  template,
+  styles,
+  shadowOptions: { mode: "open" },
+})
+export default class TESGetRun extends FASTElement {
+  @attr id = "";
+
+  @attr state = "";
+
+  @observable isLoading = true;
```

Comment:
is this some sort of safety check? And is there a situation where the component calls it's connected callback while this evaluates to false?

Reply:
isLoading is just to render skeleton/loading animation whenever new data needs to be fetched.

Reply:
Yeah but you're using it to check if to add an event listener to the delete button. And it seems to me that every time this connected callback is called (the component will have just mounted,) "isLoading" is set to true whenever this happens so that's why i asked, is there a situation where the component calls it's connected callback while "isLoading" evaluates to false?